syntax = "proto3";

package saq.sandwich.proto;


// Tunnel modes.
enum Mode {
  // Unspecified mode.
  MODE_UNSPECIFIED = 0;
  // Client mode.
  MODE_CLIENT = 1;
  // Server mode.
  MODE_SERVER = 2;
}

// Events that can be polled from a tunnel.
// Multiple events can be bitwise-OR by `sandwich_tunnel_poll`.
enum PollEvents {
  // No-op option.
  POLL_UNSPECIFIED = 0;

  // Poll is not supported.
  POLL_UNSUPPORTED = 1;

  // There is data to read.
  POLL_IN = 2;

  // Writing is possible.
  POLL_OUT = 4;

  // The remote peer closed the connection.
  POLL_HUP = 8;

  // An error occurred.
  POLL_ERR = 16;
}

// Information that can be obtained from a tunnel.
enum TunnelInfo {
  // The network address (string).
  TUNNEL_INFO_NETADDRESS = 0;

  // The network port (integer)..
  TUNNEL_INFO_NETPORT = 1;
}
